### PR TITLE
Pretrained weights for detection-light and segmentation-light

### PR DESCRIPTION
**Pull request by klemen1999**
## Pretrained weights for detection-light and segmentation-light

This PR adds option to download pretrained weights for detection and segmentation predefined models (light versions). Checkpoints are split per node and saved in the release in luxonis-train ([v0.1.0](https://github.com/luxonis/luxonis-train/releases/tag/v0.1.0-beta)). If they are not present at the start of the training then we automatically download them otherwise we used the cached ones. 
By default we set `download_weights` to `True` for all backbones (where applicable) and if using predefined models we also download them for neck and head nodes.

### CODE REVIEW

File: luxonis_train/config/predefined_models/detection_model.py [hunk 1]
```diff
@@ -27,10 +29,14 @@ def get_variant(variant: VariantLiteral) -> DetectionVariant:
...
             backbone="EfficientRep",
             backbone_params={"variant": "n"},
+            neck_params={"variant": "n", "download_weights": True},
+            head_params={"download_weights": True},
         ),
         "heavy": DetectionVariant(
             backbone="EfficientRep",
             backbone_params={"variant": "l"},
+            neck_params={"variant": "l", "download_weights": True},
+            head_params={"download_weights": False},
```
Comment (sokovninn) [inline luxonis_train/config/predefined_models/detection_model.py:39]: We have a pretrained backbone for the heavy detection model, but not a pretrained head, right?

File: configs/detection_light_model.yaml [hunk 2]
```diff
@@ -31,15 +32,21 @@ trainer:
   optimizer:
     name: SGD
     params:
-      lr: 0.02
-      momentum: 0.937
-      weight_decay: 0.0005
-      dampening: 0.0
-      nesterov: true
+      lr: 0.0032
```
Comment (kozlov721) [inline configs/detection_light_model.yaml:35]: Just out of curiosity where do these values come from?

**Review by sokovninn — APPROVED** [review]
LGTM, left a few comments.

File: luxonis_train/config/predefined_models/detection_model.py (hunk 1 above)
Comment (klemen1999) [reply luxonis_train/config/predefined_models/detection_model.py:39]: Not yet. We've only trained light detection model on COCO so we only have "nano" variant for the backbone, neck and head. We'll be adding more gradually, as we go.

File: luxonis_train/nodes/README.md [hunk 3]
```diff
@@ -53,7 +53,7 @@ Adapted from [here](https://pytorch.org/vision/main/models/resnet.html).
 | Key                | Type                                      | Default value | Description                            |
 | ------------------ | ----------------------------------------- | ------------- | -------------------------------------- |
 | `variant`          | `Literal["18", "34", "50", "101", "152"]` | `"18"`        | Variant of the network                 |
-| `download_weights` | `bool`                                    | `False`       | If True download weights from ImageNet |
+| `download_weights` | `bool`                                    | `True`        | If True download weights from ImageNet |
```
[the remark this answers is not in the document]
Comment (klemen1999) [reply luxonis_train/nodes/README.md:56]: Yes eventually, specially for the "whole models" (e.g. predefined detection or predefined segmentation model"

File: configs/detection_light_model.yaml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 # Example configuration for training a predefined light detection model
+# NOTE: This example downloads pretrained COCO weights and training parameters are already prepared for fine tuning
```
[the remark this answers is not in the document]
Comment (klemen1999) [reply configs/detection_light_model.yaml:2]: Currently we allow loading predefined weights for specific module (e.g. our COCO weights for EfficientRep) and we also allow loading custom weights for the whole architecture through `model.weights` in YAML. 
IMO it's too early to allow loading specific checkpoints for each Node - by too early I mean that I don't see a use case for it. But if there is we can add this as well quite easily.

File: luxonis_train/utils/general.py [hunk 5]
```diff
@@ -139,3 +143,63 @@ def get_with_default(
...
+                if i == retry:
+                    logger.warning("Download failed, retry limit reached.")
+                    return None
+                logger.warning(f"Download failed, retrying {i+1}/{retry} ...")
+
+
+def clean_url(url: str) -> str:
+    """Strip auth from URL, i.e. https://url.com/file.txt?auth -> https://url.com/file.txt."""
+    url = (
+        Path(url).as_posix().replace(":/", "://")
```
Comment (kozlov721) [inline luxonis_train/utils/general.py:196]: You can use `PurePosixPath` instead of `Path` here so you don't have to deal with Windows

**Review by kozlov721 — APPROVED** [review]
Left a small suggestion, otherwise LGTM

File: configs/detection_light_model.yaml (hunk 2 above)
Comment (klemen1999) [reply configs/detection_light_model.yaml:35]: Official YoloV6 finetune config

File: luxonis_train/config/predefined_models/detection_model.py (hunk 1 above)
Comment (sokovninn) [reply luxonis_train/config/predefined_models/detection_model.py:39]: Ok, but why do we have `"download_weights": True` here in the heavy variant?

Comment (klemen1999) [reply luxonis_train/config/predefined_models/detection_model.py:39]: Oh yeah nice catch, will fix.

File: luxonis_train/config/predefined_models/detection_model.py [hunk 6]
```diff
@@ -27,10 +29,14 @@ def get_variant(variant: VariantLiteral) -> DetectionVariant:
         "light": DetectionVariant(
             backbone="EfficientRep",
             backbone_params={"variant": "n"},
+            neck_params={"variant": "n", "download_weights": True},
+            head_params={"download_weights": True},
         ),
         "heavy": DetectionVariant(
             backbone="EfficientRep",
             backbone_params={"variant": "l"},
+            neck_params={"variant": "l", "download_weights": True},
```
Comment (klemen1999) [inline luxonis_train/config/predefined_models/detection_model.py:38]:
```suggestion
            neck_params={"variant": "l", "download_weights": False},
```

**Pull request merged**